src/config/azureKeyVaultClient.ts

```diff
@@ -0,0 +1,16 @@
+import { Config } from './environmentConfig';
+import { DefaultAzureCredential } from '@azure/identity';
+import { SecretClient } from '@azure/keyvault-secrets';
+
+/**
+ * Function: getKeyVaultClient
+ *
+ * This function returns a new instance of SecretClient from the '@azure/keyvault-secrets' package.
+ * It uses the Config object from './environmentConfig' to get the azureKeyVaultURL and the DefaultAzureCredential
+ * from the '@azure/identity' package to authenticate the client.
+ *
+ * @returns {SecretClient} - A new instance of SecretClient.
+ */
+const getKeyVaultClient = () => new SecretClient(Config.azureKeyVaultURL!, new DefaultAzureCredential());
```

Comment:
please make it as singleton pattern. we don't need to pass new instance every time we request the azure secret client

Reply:
Yes, now we only use it if there is not an instance available for use.